src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java

```diff
@@ -170,10 +164,8 @@ private String extractRequestBody(StaplerRequest req) {
         return body;
     }
 
-    
-    private Set<GhprbWebHook> getWebHooks() {
-        final Set<GhprbWebHook> webHooks = new HashSet<GhprbWebHook>();
-
+    private Set<GhprbTrigger> getTriggers(String repoName, String body, String signature) {
+        final Set<GhprbTrigger> triggers = new HashSet<GhprbTrigger>(5);
```

Comment:
That would be fantastic.  What are the implications around threading, addition and removal of triggers and jobs, etc.?  It is possible to keep the map up to date?

Really I think you want a multi hash map from repo name -> triggers.